### ISSUE TITLE
Optimistic rebuild in case of surplus events found

### ISSUE THREAD

**Issue by timbru** [NLnetLabs/krill#530]
Krill falls back to 'recover' mode in case surplus events are found at startup when the cache is 'warmed'.

In itself this is fine, however, it does a full recovery from scratch (event 0 to now) in this case.

Surplus event are likely to be caused by a full disk mid 'transaction'. And since Krill crashes on I/O errors the `snapshot.json` and `info.json` files would not have been changed - they are written last. Furthermore, older delta files are never overwritten.

In conclusion, we can and should be a bit more optimistic. Meaning: just archive the surplus events and command if applicable, and then continue. I.e. throw away the last incomplete transaction only.

If that fails we can still do a full recovery.
